amazon-kinesis-client/src/test/java/software/amazon/kinesis/common/StreamIdentifierTest.java

```diff
@@ -51,72 +41,88 @@ public void testMultiStreamDeserializationSuccess() {
-                "arn:aws:kinesis:region::stream/stream-name", // missing account id
-                "arn:aws:kinesis:region:123456789:stream/stream-name", // account id not 12 digits
-                "arn:aws:kinesis:region:123456789abc:stream/stream-name", // 12char alphanumeric account id
-                "arn:aws:kinesis:region:123456789012:stream/", // missing stream-name
-                // serialization examples
                 ":stream-name:123", // missing account id
 //                "123456789:stream-name:123", // account id not 12 digits
                 "123456789abc:stream-name:123", // 12char alphanumeric account id
                 "123456789012::123", // missing stream name
                 "123456789012:stream-name", // missing delimiter and creation epoch
                 "123456789012:stream-name:", // missing creation epoch
+                "123456789012:stream-name:-123", // negative creation epoch
                 "123456789012:stream-name:abc", // non-numeric creation epoch
                 ""
         )) {
             try {
                 StreamIdentifier.multiStreamInstance(pattern);
-                Assert.fail(pattern + " should not have created a StreamIdentifier");
+                Assert.fail("Serialization " + pattern + " should not have created a StreamIdentifier");
             } catch (final IllegalArgumentException iae) {
                 // expected; ignore
             }
         }
     }
 
+    /**
+     * Test Arns that <b>should not</b> match a valid AWS Kinesis stream Arn.
+     */
     @Test
-    public void testInstanceFromArn() {
-        final Arn arn = toArn(KINESIS_REGION);
-        final StreamIdentifier single = StreamIdentifier.singleStreamInstance(arn.toString());
-        final StreamIdentifier multi = StreamIdentifier.multiStreamInstance(arn.toString());
-
-        assertEquals(single, multi);
-        assertEquals(Optional.empty(), single.streamCreationEpochOptional());
-        assertActualStreamIdentifierExpected(arn, single);
+    public void testMultiStreamByArnWithInvalidStreamArnFail() {
+        final String region = KINESIS_REGION.id();
+        for (final Arn invalidStreamArn : Arrays.asList(
+                streamArn("abc", SERVICE,  region, TEST_ACCOUNT_ID, RESOURCE), // invalid partition
+                streamArn(PARTITION, "dynamodb",  region, TEST_ACCOUNT_ID, RESOURCE), // incorrect service
+                streamArn(PARTITION, SERVICE,  null, TEST_ACCOUNT_ID, RESOURCE), // missing region
+                streamArn(PARTITION, SERVICE,  region, null, RESOURCE), // missing account id
+                streamArn(PARTITION, SERVICE,  region, "123456789", RESOURCE), // account id not 12 digits
+                streamArn(PARTITION, SERVICE,  region, "123456789abc", RESOURCE), // 12char alphanumeric account id
+                streamArn(PARTITION, SERVICE,  region, TEST_ACCOUNT_ID, "table/name"), // incorrect resource type
```

Comment:
Would you kindly include the following (if-and-only-if the ARN builder allows):
1. A Kinesis ARN w/ empty resource (i.e., `""`).
1. A Kinesis ARN w/o a stream name (i.e., `stream/`).

Reply:
ARN builder does not allow these cases.